pkgs/top-level/all-packages.nix

```diff
@@ -12524,7 +12524,9 @@ in
 
   libxmi = callPackage ../development/libraries/libxmi { };
 
-  libxml2 = callPackage ../development/libraries/libxml2 { };
+  libxml2 = callPackage ../development/libraries/libxml2 {
+    python3 = if stdenv.isDarwin then python2 else python3;
```

Comment:
Passing `python2` via a function argument named `python3` is super confusing. Better to just call the function argument `python`. Even better is to make it work the same on macOS...